apache/superset
# feat: Adds drill to detail context menu for ECharts visualizations

### SUMMARY
Adds drill to detail (drill-through) context menu for ECharts visualizations. The context menu is triggered when right-clicking an EChart series. Upon selecting a drill to detail option, an alert is displayed with the selected values.

The linking between the context menu and the actual feature will be handled in a future PR.

The context menu was added to all ECharts with the exception of the Tree Chart.

The feature is behind the `DRILL_TO_DETAIL` feature flag.

This first iteration focuses on providing a minimal set of interaction possibilities for the ECharts. More advanced interactions like clicking on legends, subheaders, etc will be handled in follow-up PRs. Drilling to detail using metrics is not supported as well.

@kasiazjc @lauderbaugh @rusackas @codyml 

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
https://user-images.githubusercontent.com/70410625/181340086-a6a8376f-e962-440c-b5f4-8679bc731337.mov

### TESTING INSTRUCTIONS

1 - Make sure the feature does not introduce any regression when disabled. This is the default value for the feature flag.

2 - Enable the `DRILL_TO_DETAIL` feature flag and check:
- Check if there are no regressions when right-clicking the series for different charts
- Check if the right-click works with different chart control values (dimensions, time grain, customizations, etc)
- Check if the correct values are displayed in the context menu
- Check if the correct values are displayed when selecting a context menu option
- Check if other dashboard interactions are not affected by the new feature such as cross-filtering, drag-and-drop, tooltips, etc

Below is the import file for the dashboard shown in the video:
[dashboard_export_20220727T144027.zip](https://github.com/apache/superset/files/9202425/dashboard_export_20220727T144027.zip)

### ADDITIONAL INFORMATION
- [x] Required feature flags: `DRILL_TO_DETAIL`
